hummingbot/connector/gateway/clob_spot/data_sources/xrpl/xrpl_api_data_source.py

```diff
@@ -80,6 +85,40 @@ async def start(self):
     async def stop(self):
         await super().stop()
 
+    async def _update_markets(self):
+        # TODO: This is very similar to the lock implementation in the base class. Although it is a naive attempt to fix
+        #       the issue where checks in status_dict get delayed a lot making connector to load very slowly, we should
+        #       consider refactoring this in the next iteration to avoid code duplication. This is a temporary fix.
+        # TODO: Implement a TTL cache for this because at init time,  symbols_mapping, trading_rule, and user_strean are
+        #       all calling this at the same time, making a bottle neck thus delaying the connector to load.
+        await self._loading_markets_event.wait()
+
+        self._loading_markets_event.clear()
+        try:
+            for market_info in await self._get_markets_info():
+                trading_pair = self._get_trading_pair_from_market_info(market_info=market_info)
+                self._markets_info[trading_pair] = market_info
+                base, quote = split_hb_trading_pair(trading_pair=trading_pair)
+                base_exchange, quote_exchange = self._get_exchange_base_quote_tokens_from_market_info(
+                    market_info=market_info
+                )
+                self._hb_to_exchange_tokens_map[base] = base_exchange
+                self._hb_to_exchange_tokens_map[quote] = quote_exchange
+        finally:
```

Comment:
probably adding a except and logging the error will be good to understand what is happening